doc/source/how-to/setup.rst

```diff
@@ -0,0 +1,451 @@
+
+    .. group-tab:: Windows PowerShell
+
+        .. code-block:: text
+
+            git --version
+
+    .. group-tab:: macOS
+
+        .. code-block:: text
+
+            git --version
+
+    .. group-tab:: Linux/UNIX
+
+        .. code-block:: text
+
+            git --version
+
+Usage
+~~~~~
+If you are not familiar with this tool and would like to know how to use it,
+please refer to the `Git Reference Manual <https://git-scm.com/doc>`_. In
+previous website, you will find plenty of resources for learning Git.
+
+Configuration
+~~~~~~~~~~~~~
+It is very important to properly configure Git so every modification you made to
+the code properly points towards you. There are two types of configuration:
+:ref:`Global` and :ref:`Local`. It is also possible to combine both and have
+:ref:`Multiple` configurations. 
+
+Global
+++++++
+Global configuration will be automatically included in ever Git repository of
+your machine unless override by a :ref:`Local` configuration. This configuration
+is located in ``C:\Users\<username>\.gitconfig`` for Windows users or in
+``/home/<username>/.gitconfig`` for macOS/Linux/UNIX ones.
+
+You can set the value for any variable in a field by running:
+
+.. code-block:: bash
+
+   git config --global <field>.<varname> <value>
+
+Some examples are provided in the lines below:
+
+**Setup your name:** 
+
+.. code-block:: bash
+
+    git config --global user.name <Your Name>
+
+**Setup your email:** 
+
+.. code-block:: bash
+
+    git config --global user.email <Ansys Email>
+
+**Setup default branch name:** 
+
+.. code-block:: bash
+
+    git config --global init.defaultBranch main
+
+Local
++++++
+...
+
+Multiple
+++++++++
+The dynamic configuration allows you to have a 
+
+
+Signing Commits
+~~~~~~~~~~~~~~~
+.. todo::
+
+Enabling SSH
+~~~~~~~~~~~~
+.. todo::
+
+
+WSL2
+----
+Some developers prefer using Windows as the operative system for their machines.
+However, they may like to take advantage from some features provided by a Linux
+system. To solve for this problem, the `Windows Subsystem for Linux
+<https://docs.microsoft.com/en-us/windows/wsl/install>`_ was devised.
+
+Installation
+~~~~~~~~~~~~
+Open a new PowerShell session and enter the following command:
+
+.. code-block:: powershell
+
+   wsl --install
+
+After installing ``WSL``, make sure you are running the ``WSL2`` version by
+executing:
+
+.. code-block:: powershell
+
+   wsl --set-default-version 2
+
+Verifying Installation
+~~~~~~~~~~~~~~~~~~~~~~
+To check which ``WSL`` version are you using, run:
+
+.. code-block:: powershell
+
+   wsl --list -v
+
+Installing a Linux Distro
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Once ``WSL2`` has been installed, it is time to install a Linux
+distribution. To get a list of the available ones, run:
+
+.. code-block:: powershell
+
+   wsl --list --online
+
+Most developers choose `Ubuntu <https://ubuntu.com/download>`_, as it is a
+heavily maintained Linux distribution with a huge collection of packages.
```

Comment:
```suggestion
Most developers choose `Ubuntu <https://ubuntu.com/download>`_ because it is a
well maintained Linux distribution with a huge collection of packages.
```